tests/pages/LoginPage.ts

```diff
@@ -0,0 +1,82 @@
+import { expect, type Locator, type Page } from "@playwright/test";
+import { BasePage } from "./BasePage";
+
+/**
+ * Login Page Object - Represents the user login page
+ */
+export class LoginPage extends BasePage {
+  readonly welcomeBackHeading: Locator;
+  readonly emailField: Locator;
+  readonly passwordField: Locator;
+  readonly submitButton: Locator;
+  readonly logoutButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.welcomeBackHeading = page.getByRole("heading", { name: "Welcome Back" });
+    this.emailField = page.locator("#email");
+    this.passwordField = page.locator("#password");
+    this.submitButton = page.getByRole("button", { name: /Login/i });
+    this.logoutButton = page.getByRole("button", { name: /Logout/i });
+  }
+
+  async open() {
+    await this.goto("/login");
+  }
+
+  async verifyUrl() {
+    await expect(this.page).toHaveURL(/\/login/);
+  }
+
+  async verifyHeading() {
+    await expect(this.welcomeBackHeading).toBeVisible();
+  }
+
+  async verifyFormFields() {
+    await expect(this.page.getByLabel(/Email/i)).toBeVisible();
+    await expect(this.page.getByLabel(/Password/i).first()).toBeVisible();
+  }
+
+  generateUniqueEmail(): string {
+    const timestamp = Date.now();
+    return `testuser${timestamp}@example.com`;
```

Comment:
Not something to change - but be careful with things like this in tests. 

The timestamp at this comment is 1762447749, 10 seconds have now passed and its 1762447759, assertions could fail with logic like this if the timestamp is ever regenerated within the same test